docs/eBpfExtensions.md

```diff
@@ -189,7 +199,14 @@ This optional function is used to populate the flat buffers representing the dat
 returned to the application when the `bpf_prog_test_run_opts` call completes. In addition, the function frees any
 resources allocated in the `ebpf_program_context_create_t` call.
 
-### 2.2 Program Information NPI Client Attach and Detach Callbacks
+### 2.2 Backward compatibility of the Extension data structures
+All the extension data structures are versioned. New fields can be added to the end of a data structure to maintain backward compatibility
+with the existing extensions. In such cases, the size field of the header will be updated but the version of the structure *will not change*.
+If the change in data structure is such that it is no longer backward compatible (such as change field type or position),
+then the version number will be updated. All previous versions of extension structure will continue to be supported.
```

Comment:
```suggestion
then the version number will be updated. All previous versions of the extension structure will continue to be supported.
```

Reply:
I modified the text here. Version numbers are updated for breaking changes. Older types will no longer be supported.